entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -236,6 +244,22 @@ class EntityHandler(
             ResponseEntity.status(HttpStatus.MULTI_STATUS).body(updateResult)
     }
 
+    // TODO add support for multi attribute
+    private fun extractDatasetIdFromNgsiLdAttributes(
+        ngsiLdAttributes: List<NgsiLdAttribute>,
+        attributeName: String
+    ): URI? {
+        val ngsiLdAttribute = ngsiLdAttributes.first {
+            it.name == attributeName
+        }
+        return when (ngsiLdAttribute) {
```

Comment:
or you could add a function `fun getAttributeInstances(): List<NgsiLdAttributeInstance>` in NgsiLdAttribute (that would simply return the `instances` member in each implementing class. There in this function, you can just do `ngsiLdAttributes.first {
            it.name == attributeName
        }.getAttributeInstances()[0].datasetId`